engine/src/io/data_parser/sql/MySQLParser.cpp

```diff
@@ -0,0 +1,598 @@
+        host_cols[col] = tmp;
+      } break;
+      case cudf::type_id::INT16: {
+        std::vector<int16_t> *tmp = new std::vector<int16_t>();
+        tmp->resize(total_rows);
+        host_cols[col] = tmp;
+      } break;
+      case cudf::type_id::INT32: {
+        std::vector<int32_t> *tmp = new std::vector<int32_t>();
+        tmp->resize(total_rows);
+        host_cols[col] = tmp;
+      } break;
+      case cudf::type_id::INT64: {
+        std::vector<int64_t> *tmp = new std::vector<int64_t>();
+        tmp->resize(total_rows);
+        host_cols[col] = tmp;
+      } break;
+      case cudf::type_id::UINT8: {
+        std::vector<uint8_t> *tmp = new std::vector<uint8_t>();
+        tmp->resize(total_rows);
+        host_cols[col] = tmp;
+      } break;
+      case cudf::type_id::UINT16: {
+        std::vector<uint16_t> *tmp = new std::vector<uint16_t>();
+        tmp->resize(total_rows);
+        host_cols[col] = tmp;
+      } break;
+      case cudf::type_id::UINT32: {
+        std::vector<uint32_t> *tmp = new std::vector<uint32_t>();
+        tmp->resize(total_rows);
+        host_cols[col] = tmp;
+      } break;
+      case cudf::type_id::UINT64: {
+        std::vector<uint64_t> *tmp = new std::vector<uint64_t>();
+        tmp->resize(total_rows);
+        host_cols[col] = tmp;
+      } break;
+      case cudf::type_id::FLOAT32: {
+        std::vector<float> *tmp = new std::vector<float>();
+        tmp->resize(total_rows);
+        host_cols[col] = tmp;
+      } break;
+      case cudf::type_id::FLOAT64: {
+        std::vector<double> *tmp = new std::vector<double>();
+        tmp->resize(total_rows);
+        host_cols[col] = tmp;
+      } break;
+      case cudf::type_id::BOOL8: {
+        std::vector<int8_t> *tmp = new std::vector<int8_t>();
+        tmp->resize(total_rows);
+        host_cols[col] = tmp;
+      } break;
+      case cudf::type_id::TIMESTAMP_DAYS:
+      case cudf::type_id::TIMESTAMP_SECONDS:
+      case cudf::type_id::TIMESTAMP_MILLISECONDS:
+      case cudf::type_id::TIMESTAMP_MICROSECONDS:
+      case cudf::type_id::TIMESTAMP_NANOSECONDS:
+      // TODO percy it seems we don't support this yet
+      // case cudf::type_id::DURATION_DAYS: {} break;
+      // case cudf::type_id::DURATION_SECONDS: {} break;
+      // case cudf::type_id::DURATION_MILLISECONDS: {} break;
+      // case cudf::type_id::DURATION_MICROSECONDS: {} break;
+      // case cudf::type_id::DURATION_NANOSECONDS: {} break;
+      // case cudf::type_id::DICTIONARY32: {} break;
+      case cudf::type_id::STRING: {
+        auto *tmp = new cudf_string_col();
+        tmp->offsets.resize(1, 0);
+        host_cols[col] = tmp;
+      } break;
+      // TODO percy
+      case cudf::type_id::LIST: {} break;
+      case cudf::type_id::DECIMAL32: {} break;
+      case cudf::type_id::DECIMAL64: {} break;
+      case cudf::type_id::STRUCT: {} break;
+      case cudf::type_id::NUM_TYPE_IDS: {} break;
+    }
+  }
+
+  int row = 0;
+  while (res->next()) {
```

Comment:
can mysql c api only get one row at a time?

Reply:
With regards to the range checking, if we have a value outside the range, we would just have to throw an error. We cant change the type midway. But that is better than nothing. I think we will implement Unsigned types soon, so this will hopefully be a short term issue

Reply:
@felipeblazing yes unfortunately mysql is using the jdbc approach and there is nothing like fetch_bacth in the jdbc standard, I didn't find anything like "fetch_batch" in the native C neither. The good news is that I think the underlying implementation of those drivers have a buffer of results and the client API is just returning the latest item that was fetched.